java/vector/src/main/codegen/templates/AbstractFieldWriter.java

```diff
@@ -98,6 +98,12 @@ public ListWriter list() {
   }
 
   @Override
+  public ListWriter list(int size) {
+    fail("FixedSizeList");
+    return null;
+  }
```

Comment:
let's hold off adding the fixed width types to the writer APIs.
One of the goals of those APIs is to be able to ingest arbitrary data (for example JSON)  and infer the schema on the fly (hence the PromotableWriter promoting vectors to union if the type is not homogenous). So it won't work well with fixed width list since nothing prevents having more than one size in this api.